### PR TITLE
Update Readme and docs

### PR DESCRIPTION
**Pull request by plettich**
- Add information about stamping the database
- Add information about the documentation to CONTRIBUTING
- Update README, removed part about submodules, minor flaws

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -98,35 +101,31 @@ If you want to setup a development environment start like this::
...
-Create the database and encryption key::
+First You need to create a `config-file <https://privacyidea.readthedocs
+.io/en/latest/installation/system/inifile.html>`_.
+
+Then create the database and encryption key::
 
     ./pi-manage createdb
     ./pi-manage create_enckey
 
+If You plan to use the database for a longer time, You should `stamp <https://privacyidea
```
Comment (cornelinux) [inline README.rst]: "for a longer time" sounds a bit strange to me. I know what you mean: Stamping is not necessary if I simply use the system like "now", within one release, fix a bug, do a PR and delete my whole dev folder.

I am not sure how to say this. Maybe s.th. like "If you want to have the development database to be upgradeable you need to stamp it like this::"

File: README.rst [hunk 2]
```diff
@@ -145,9 +144,11 @@ as administrator.
 Run tests
 =========
 
-If you have followed the steps above to set up your environment for testing, running the test suite should be as easy as running nosetests(1) with the following options::
+If you have followed the steps above to set up your environment for testing,
+running the test suite should be as easy as running
+`pytest <http://pytest.org/>`_ with the following options::
```
Comment (cornelinux) [inline README.rst]: I think in the environment we would also have to install pytest. Run 

    pip install pytest

(I think it was not installed before)

File: README.rst (hunk 1 above)
Comment (plettich) [reply README.rst]: Fixed in 00d3330

File: README.rst (hunk 2 above)
Comment (plettich) [reply README.rst]: actually it is part of `tests/requirements.txt`. I added a link to the installation of the testing environment in 00d3330

**Pull request merged**